Trash manager

alroniks opened the pull request:
See original PR here https://github.com/modxcms/revolution/pull/13764 for details.

This is rebased version with additional improvements. Original will be closed after finish if work on improvements.

## Template build required
The PR needs to run `_build/templates/default/grunt build`. Otherwise the hover state of the buttons is not visible and the changes in modx.tree.resource.js are not in the compressed scripts.

Commit:
Restored from broken branch.

Commit:
Removed wrong file.

Commit:
Added missing file.

Commit:
Added sort to access policy.

Commit:
Removed wrong tempfile.

Commit:
Added policy permission to show the trash manager menu.

Commit:
Fixed bugs with single resource restore.
Added lexicon strings.

Commit:
Linked recycle bin icon to trash manager page.

Commit:
Fixed logging messages.

Commit:
Code cleanup.

Commit:
Implemented code review advices.

Commit:
Fixed lexicon file formatting

Commit:
Undone useless changes and removed commented lines

Commit:
Fixed code formatting to continue with clean code

Commit:
Splitted panel to grid and panel and some refactoring

JoshuaLuckers commented on core/model/modx/processors/resource/trash/restore.class.php:
```diff
@@ -0,0 +1,196 @@
+                $this->success[] = $id;
+            }
+
+            // TODO this still has to be discussed: what happens to the children?
```
Issue #11973 might be sort of relevant here

Jako committed:
Some refactorings
- Trash icon tooltip is updated on resource delete/undelete
- Trash icon is clickable now
- Pagetitle column tooltip contains some resource informations
- Search/Clear works now
- Manager controller moved to resource/trash
- Lexicon issues
- Button hover colors

alroniks committed:
Some refactoring from Jako

Jako committed:
Context selector and CSS improvement

Jako commented:
Checked, refactored, working fine.

Jako committed:
Exclude mgr context from combo

alroniks commented:
I also want to do some refactoring in grid component, but right now I have a lack of time. So I removed state `ready-for-merging`

Mark-H reviewed (approved):
While it looks like there's still work being done, I'll go ahead and approve the current state. Looks really promising.

opengeek commented:
Is this ready for merge or not? It's approved for it but says it needs review and @Mark-H mentioned it looks like work is still ongoing? @Alroniks ?

alroniks commented:
I wanted to improve some in code, fix some smelling places, but it works well and can be merged, all further improvements I can add later. So feel free to merge. I don't want to block a release.
